### PR TITLE
fixes for Melpa

### PR DESCRIPTION
**Pull request by NicoloZorzetto**
Hello, these are the fixes they asked. I had to remove the setting of the font and they asked about some sets implying a theme shouldn't do that. I think we can get away with it but if they really won't let us I'm afraid we'll need to make a minor mode that sets the modeline etc and color themes to set the colors.

Again, I think we can get away with it but time will tell.

### CODE REVIEW

**Comment by NicoloZorzetto** [comment]
They also asked for a GitHub compatible license:

`Add a LICENSE file that GitHub can detect (e.g. no markup) if possible
See: https://github.com/licensee/licensee`

File: elegant-emacs-common.el [hunk 1]
```diff
@@ -1,4 +1,4 @@
-;;; elegant-emacs-light-theme.el --- A very minimal but elegant and consistent theme
+;;; elegant-emacs-common.el --- A very minimal but elegant and consistent theme
 ;;; Copyright (C) 2020 Nicolas P. Rougier
```
Comment (rougier) [inline elegant-emacs-common.el]: You should add  your name here

File: elegant-emacs-common.el [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 ;;; -------------------------------------------------------------------
 ;;; Author: Nicolas P. Rougier
```
Comment (rougier) [inline elegant-emacs-common.el]: you should add your name here

File: elegant-emacs-common.el [hunk 3]
```diff
@@ -70,7 +70,7 @@
 
 ;;; When we set a face, we take care of removing any previous settings
 ;;; -------------------------------------------------------------------
-(defun set-face (face style)
+(defun elegant-emacs-common-set-face (face style)
```
Comment (rougier) [inline elegant-emacs-common.el]: If we rename the file `elegant.el`, then we could use the `elegant-` prefix everywhere, what do you think ?

File: elegant-emacs-common.el [hunk 4]
```diff
@@ -142,12 +142,12 @@ background color that is barely perceptible."
 ;;; Mode line rendering
 ;;; -------------------------------------------------------------------
 ;;; This line below makes things a bit faster
-(set-fontset-font "fontset-default"  '(#x2600 . #x26ff) "Fira Code 16")
+;;; (set-fontset-font "fontset-default"  '(#x2600 . #x26ff) "Fira Code 16")
```
Comment (rougier) [inline elegant-emacs-common.el]: Not sure how to replace this line which is not totally mandatory but make things faster. This is actually used to set the ↩︎ symbol using Fira Code. If this is not set, Emacs needs to look for a font where this symbol is defind and it tends to make things slow (on my machin at least).

File: elegant-emacs-dark-theme.el [hunk 5]
```diff
@@ -1,4 +1,4 @@
-;;; elegant-emacs-light-theme.el --- A very minimal but elegant and consistent theme
+;;; elegant-emacs-dark-theme.el --- A very minimal but elegant and consistent theme
```
Comment (rougier) [inline elegant-emacs-dark-theme.el]: `elegant-dark.el` instead?

File: elegant-emacs-common.el (hunk 4 above)
Comment (NicoloZorzetto) [reply elegant-emacs-common.el]: Yes that's what I figured but in the Melpa pull they say that it's not something a theme should do. This is a quick fix, once approved the fetchers will look for the files so we can do anything to them apart from renaming

File: elegant-emacs-common.el (hunk 3 above)
Comment (NicoloZorzetto) [reply elegant-emacs-common.el]: Yea, could be done if it's ok with you. Where did you want to add the prefix?

File: elegant-dark-theme.el [hunk 6]
```diff
@@ -0,0 +1,74 @@
+;;; elegant-dark-theme.el --- A very minimal but elegant and consistent theme
+;;; Copyright (C) 2020 Nicolas P. Rougier and Nicolò Zorzetto
+;;; -------------------------------------------------------------------
+;;; Authors: Nicolas P. Rougier and Nicolò Zorzetto
```
Comment (NicoloZorzetto) [inline elegant-dark-theme.el:4]: Is my name here ok?

File: elegant-dark-theme.el [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+;;; -------------------------------------------------------------------
+;;; URL: https://github.com/rougier/elegant-emacs
+;;; -------------------------------------------------------------------
+;;; Version: 1
```
Comment (NicoloZorzetto) [inline elegant-dark-theme.el:8]: I changed the version to `1`, maybe `1.0` is better? Or did you want to keep `0.1` ?

**Comment by NicoloZorzetto** [comment]
I changed the names like you suggested, how are we going to do the copyright thing?

File: elegant-dark-theme.el (hunk 7 above)
Comment (rougier) [reply elegant-dark-theme.el:8]: No strong opinion on that. Maybe 1.0 if we add minor changes.

File: elegant-dark-theme.el (hunk 6 above)
Comment (rougier) [reply elegant-dark-theme.el:4]: Yes !

**Comment by rougier** [comment]
Hum, I cannot merge because of conflicts but GitHub is not very helpful.

**Comment by NicoloZorzetto** [comment]
It is in conflict because I renamed the files and it’s detecting them as deleted. Do you prefer I add the old ones so you can merge and then delete them?

**Comment by rougier** [comment]
Well, I cannot merge nor resolve from the GitHub interface, weird. I'll give a try from terminal

**Comment by rougier** [comment]
It worked.

**Comment by NicoloZorzetto** [comment]
Nice! Now only thing remaining is the copyright. Should I look into that?

**Comment by rougier** [comment]
Oh yes. Also, we can get rid of `sanity.el` and `elegance.el`

**Comment by rougier** [comment]
By license, you mean adding the GPL3 in the repo ?

**Comment by NicoloZorzetto** [comment]
They asked for this:


Add a LICENSE file that GitHub can detect (e.g. no markup) if possible
See: https://github.com/licensee/licensee

**Comment by rougier** [comment]
Just added LICENSE.txt to the repo.

**Comment by rougier** [comment]
And made some adjustements to the layout (underline descent and header line right part alignment)

**Comment by NicoloZorzetto** [comment]
Perfect! I'll let them know

**Comment by NicoloZorzetto** [comment]
They  **seem** to like the recipe although I initially forgot to change the name of the files.

I was wondering: do you want to change the name of the package itself since we changed that of the files? I know we said to keep it so people could find it easily but maybe creating another repo redirecting to the one with the proper name could help.
The guy who's following our pull request has hinted to a change of name but I don't think it'll be a problem if you decide to keep `elegant-emacs`.

There is also the possibility of creating a minor mode but I don't know if we'd have to change the files completely or just ship `elegance.el` and the themes (without the requirement for the minor mode) separately, if you want I can look into that

**Comment by rougier** [comment]
What is the current name and that would be the alternative ? I've no strong opinion if the choice is between elegant and elegance and elegant-emacs.

**Comment by NicoloZorzetto** [comment]
Yep, I was thinking about those. I like `elegant-emacs` but I also get why someone would think it's strange to include the name emacs in that of the package.

What about the minor mode? What do you think?

**Comment by rougier** [comment]
Re-reading your question about the name: you seem to imply that changing the name would require a new repo ?
Concerning the minor mode, I'm not sure to see any advantage of having a minor mode, but I may miss something obvious.

**Comment by NicoloZorzetto** [comment]
> Re-reading your question about the name: you seem to imply that changing the name would require a new repo ?
> Concerning the minor mode, I'm not sure to see any advantage of having a minor mode, but I may miss something obvious.

Yea I was thinking of a new repo because people who know your work will search for 'elegant-emacs' and changing the name of this repo would give them no results.

I don't think having a minor mode would have many 'benefits' but I worry people at Melpa will complain seeing that a theme is changing not just the colors.

Your work is peculiar and the 'theme' section might not comprehend everything it's done here.

**Comment by NicoloZorzetto** [comment]
`I really like the visuals of this approach, but the elegant.el file is not really a package, and is more akin to a starter kit or a very wide set of hard-coded user preferences, given that it toggles all sorts of minor modes. This is typically not the sort of thing we host. The file appears to serve partly as a tool for defining the two themes (which are the sort of thing we happily host!) but it all works by using a large number of unprefixed top-level functions and variables instead of the proper custom-theme-set-faces and custom-theme-set-variables facilities, so they are not really standard Emacs themes at all. In summary, the current state of this is that it is a very interesting visual experiment, but not ready for wider distribution as a package.`

This is the last comment they left. As I worried looking into it they saw the use of top level functions and the lack of `custom-theme-set-variable`.

I don't know how to proceed, they even ruled out the possibility of a minor mode.

**Comment by rougier** [comment]
😞 That's hard. Maybe we should start a starter kit project instead. I've recenty performed some experiments with nano emacs that share a lot to thing with elegant emacs and we could try to unify them. I fear this means more work but maybe it's worth. What do you think ?

**Comment by terlar** [comment]
Perhaps it could be worth reaching out to @hlissner and the [doom emacs](https://github.com/hlissner/doom-emacs/) community. Feels like maybe there could be interest there and joining forces could provide a lot of community, expertise and reach. Just a wild suggestion :)

I might also be interested, but I am short on time for the near future.

**Comment by rougier** [comment]
In the meantime, we could use quelpa and put it forward in the readme (I did that for https://github.com/rougier/svg-tag-mode)

**Pull request merged**